sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsClientBuilder.java

```diff
@@ -0,0 +1,202 @@
+        // Closest to API goes first, closest to wire goes last.
+        List<HttpPipelinePolicy> policies = new ArrayList<>();
+
+        policies.add(new RequestIdPolicy());
+
+        HttpPolicyProviders.addBeforeRetryPolicies(policies);
+
+        policies.add(retryPolicy);
+
+        policies.add(new AddDatePolicy());
+        HttpPipelinePolicy credentialPolicy = new BearerTokenAuthenticationPolicy(tokenCredential, GetAuthorizationScopes(endpoint));
+        policies.add(credentialPolicy);
+        policies.addAll(additionalPolicies);
+        HttpPolicyProviders.addAfterRetryPolicies(policies);
+
+        policies.add(new HttpLoggingPolicy(logOptions));
+
+        return new HttpPipelineBuilder()
+            .policies(policies.toArray(new HttpPipelinePolicy[0]))
+            .httpClient(httpClient)
+            .build();
+    }
+
+    private static String[] GetAuthorizationScopes(String endpoint) {
+        // Uri representation for azure digital twin app Id "0b07f429-9f4b-4714-9392-cc5e8e80c8b0" in the public cloud.
+        String adtPublicCloudAppId = "https://digitaltwins.azure.net";
+        String defaultPermissionConsent = "/.default";
+
+        // If the endpoint is in Azure public cloud, the suffix will have "azure.net" or "ppe.net".
+        // Once ADT becomes available in other clouds, their corresponding scope has to be matched and set.
+        if (endpoint.indexOf("azure.net") > 0
+            || endpoint.indexOf("ppe.net") > 0) {
+            return new String[]{adtPublicCloudAppId + defaultPermissionConsent};
+        }
+
+        throw new IllegalArgumentException(String.format("Azure digital twins instance endpoint %s is not valid.", endpoint));
+    }
+
+    /**
+     * Create a {@link DigitalTwinsClient} based on the builder settings.
+     *
+     * @return the created synchronous DigitalTwinsClient
+     */
+    public DigitalTwinsClient buildClient() {
+        return new DigitalTwinsClient(buildAsyncClient());
+    }
+
+    /**
+     * Create a {@link DigitalTwinsAsyncClient} based on the builder settings.
+     *
+     * @return the created synchronous DigitalTwinsAsyncClient
+     */
+    public DigitalTwinsAsyncClient buildAsyncClient() {
+        Objects.requireNonNull(tokenCredential, "'tokenCredential' cannot be null.");
+        Objects.requireNonNull(endpoint, "'endpoint' cannot be null");
+
+        // Set defaults for these fields if they were not set while building the client
+        this.serviceVersion = this.serviceVersion != null ? this.serviceVersion : DigitalTwinsServiceVersion.getLatest();
+        this.retryPolicy = this.retryPolicy != null ? this.retryPolicy : new RetryPolicy(); // Default is exponential backoff
+        this.httpPipeline = this.httpPipeline != null ? httpPipeline : buildPipeline(this.tokenCredential, this.endpoint, this.logOptions, this.httpClient, this.additionalPolicies, this.retryPolicy);
```

Comment:
can we split this into multiple lines and add tab support?

Reply:
Sure, since this line in particular is a bit long